ForceSecondGod/ForceSecondGod.lua

```diff
@@ -17,55 +17,135 @@ ForceSecondGod.GodKeepsakes = {
     ForceAresBoonTrait = "AresUpgrade",
     ForceDemeterBoonTrait = "DemeterUpgrade",
 }
+ForceSecondGod.ForceNeeded = false
+ForceSecondGod.ErebusChecked = false
 
-function ForceSecondGod.IsGodForceEligible( room, godToForce, keepsakeGod, keepsakeCharges, previousOffers )
-    timesForceSeen = CurrentRun.LootTypeHistory[godToForce] or 0
+function ForceSecondGod.IsGodForceNeeded( godToForce, keepsakeGod ) -- Check if a force is needed- i.e. we don't yet have the desired number of our chosen god, and have yet to force one this room
+    local timesForceSeen = CurrentRun.LootTypeHistory[godToForce] or 0
     
+    if godToForce == keepsakeGod and CurrentRun.RunDepthCache > 14
+    or godToForce == keepsakeGod and timesForceSeen >= 2
+    or godToForce ~= keepsakeGod and timesForceSeen >= 1
+    or ReachedMaxGods() and timesForceSeen == 0
+    then
+        return false
+    end
+    DebugPrint({Text = "Second god is needed"})
+    return true
+end
+
+function ForceSecondGod.IsGodForceEligible( room, godToForce, keepsakeGod, keepsakeCharges, previousOffers ) -- If a god force is needed, check if it is eligible for a specific door
     if room.ChosenRewardType ~= "Boon"
     or not room.IsMiniBossRoom
     or Contains( previousOffers, godToForce )
     or keepsakeCharges > 0 and not Contains( previousOffers, keepsakeGod )
-    or godToForce == keepsakeGod and timesForceSeen >= 2 and CurrentRun.RunDepthCache < 15
-    or godToForce ~= keepsakeGod and timesForceSeen >= 1
-    or ReachedMaxGods() and timesForceSeen == 0
     then
         return false
     end
-    
+    DebugPrint({Text = "Forcing second god..."})
```

Comment:
I want to avoid using DebugPrint directly, just because it either does not give clear traceability (Forcing second god is clear in this case, but in other cases we have no idea what mod is logging) but it's also not reasonable to make every single mod implement a DebugPrint wrapper. I wonder if there's a way to do a stack trace in ModUtil to see which mod called a function? Then we can do something like `ModUtil.Log("...")` and have it automatically prefix `mod.ModName` or something?

I'll ping Magic about it, and if that's not feasible, we can just fall back to something like `ForceSecondGod.Log("...")`.

Reply:
Following further discussion in the discord, added this as a 1.4.0-RC1 issue (smaller release such as beta.2.1 also feasible?). Leaving DebugPrint calls in for now, but not long-term